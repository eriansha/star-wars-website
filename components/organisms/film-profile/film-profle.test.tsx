import { render } from '@testing-library/react'

import FilmProfile from './film-profile'

const props = {
  title: 'New Hope',
  openingCrawl:
    'Lorem ipsum dolor sit amet consectetur, adipisicing elit. Eligendi, itaque?',
  episodeID: 99,
  director: 'Luke Skywalker',
  producers: ['Obi Wan Kenobi']
}

describe('Film Profile', () => {
  it('renders without crashing', () => {
    render(<FilmProfile {...props} />)
  })

  it('render title props', () => {
    const { getByTestId } = render(<FilmProfile {...props} />)
    expect(getByTestId('film-title')).toHaveTextContent(props.title)
  })

  it('render director props', () => {
    const { getByTestId } = render(<FilmProfile {...props} />)
    expect(getByTestId('film-director')).toHaveTextContent(props.director)
  })

  it('render producers props', () => {
    const { getByTestId } = render(<FilmProfile {...props} />)
    expect(getByTestId('film-producers')).toHaveTextContent(
      props.producers.join('.')
    )
  })

  it('render opening crawl props', () => {
    const { getByTestId } = render(<FilmProfile {...props} />)
    expect(getByTestId('film-synopsis')).toHaveTextContent(props.openingCrawl)
  })

  it('matches snapshot', () => {
    const tree = render(<FilmProfile {...props} />)
    expect(tree.container).toMatchSnapshot()
  })
})
