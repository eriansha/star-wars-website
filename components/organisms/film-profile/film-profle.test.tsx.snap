// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Profile matches snapshot 1`] = `
<div>
  <section
    class="flex flex-col md:flex-row md:gap-x-10"
  >
    <div
      class="flex justify-center pb-5"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 250px; height: 350px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="film-poster"
          class="rounded-lg"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <div>
      <h1
        class="font-bold mb-2 text-2xl md:text-5xl"
        data-testid="film-title"
      >
        Star Wars: New Hope
      </h1>
      <div
        class="my-3"
      >
        <h3
          data-testid="film-director"
        >
          Director: Luke Skywalker
        </h3>
        <h3
          data-testid="film-producers"
        >
          Producers: Obi Wan Kenobi
        </h3>
      </div>
      <h2
        class="mb-2 text-2xl md:text-3xl"
      >
        Synopsis
      </h2>
      <p
        data-testid="film-synopsis"
      >
        Lorem ipsum dolor sit amet consectetur, adipisicing elit. Eligendi, itaque?
      </p>
    </div>
  </section>
</div>
`;
